node-tests/template_compiler_test.js

```diff
@@ -47,144 +48,309 @@ describe('TemplateCompiler', function () {
+
+        output = createBuilder(tree);
+        yield output.build();
+
+        let source = input.readText('template.hbs');
+        let expected = `export default Ember.HTMLBars.template(${htmlbarsPrecompile(source, {
+          moduleName: 'template.hbs',
+        })});`;
+        assert.strictEqual(output.readText('template.js'), expected);
+      })
+    );
+
+    it('invokes AST plugins', async function () {
+      let source = '{{foo-bar}}';
+      input.write({
+        'template.hbs': source,
+      });
+      let plugin = (env) => {
+        return {
+          name: 'fake-ast-plugin',
+
+          visitor: {
+            MustacheStatement() {
+              return env.syntax.builders.text('Huzzah!');
+            },
+          },
+        };
+      };
+
+      htmlbarsOptions.plugins.ast.push(plugin);
+
+      let tree = new TemplateCompiler(input.path(), htmlbarsOptions, true);
 
       output = createBuilder(tree);
-      yield output.build();
+      await output.build();
 
-      let source = input.readText('template.hbs');
       let expected = `export default Ember.HTMLBars.template(${htmlbarsPrecompile(source, {
         moduleName: 'template.hbs',
+        plugins: {
+          ast: [plugin],
+        },
       })});`;
-      assert.strictEqual(output.readText('template.js'), expected);
-    })
-  );
 
-  it('invokes AST plugins', async function () {
-    let source = '{{foo-bar}}';
-    input.write({
-      'template.hbs': source,
+      let outputString = output.readText('template.js');
+      assert.strictEqual(outputString, expected);
+      assert.ok(outputString.includes('Huzzah!'));
     });
-    let plugin = (env) => {
-      return {
-        name: 'fake-ast-plugin',
 
-        visitor: {
-          MustacheStatement() {
-            return env.syntax.builders.text('Huzzah!');
-          },
-        },
-      };
-    };
+    it('AST Plugins have access to `isProduction` status', async function () {
+      let source = '{{foo-bar}}';
+      input.write({
+        'template.hbs': source,
+      });
 
-    htmlbarsOptions.plugins.ast.push(plugin);
+      let wasProduction = false;
+      let plugin = (env) => {
+        wasProduction = env.isProduction;
 
-    let tree = new TemplateCompiler(input.path(), htmlbarsOptions);
+        return {
+          name: 'fake-ast-plugin',
 
-    output = createBuilder(tree);
-    await output.build();
+          visitor: {},
+        };
+      };
 
-    let expected = `export default Ember.HTMLBars.template(${htmlbarsPrecompile(source, {
-      moduleName: 'template.hbs',
-      plugins: {
-        ast: [plugin],
-      },
-    })});`;
+      htmlbarsOptions.isProduction = true;
+      htmlbarsOptions.plugins.ast.push(plugin);
 
-    let outputString = output.readText('template.js');
-    assert.strictEqual(outputString, expected);
-    assert.ok(outputString.includes('Huzzah!'));
-  });
+      let tree = new TemplateCompiler(input.path(), htmlbarsOptions, true);
```

Comment:
FWIW, adding `true` here is the thing I was referring to (as a breaking change). If you do `require('ember-cli-htmlbars')` the thing you get is the template compiler broccoli-plugin (for better or worse) which makes the constructor at least intimate API (but probably public).